graphql-dgs-spring-webmvc-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/webmvc/autoconfiguration/DgsWebMvcAutoConfigurationTest.kt

```diff
@@ -0,0 +1,128 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.webmvc.autoconfiguration
+
+import com.netflix.graphql.dgs.DgsQueryExecutor
+import com.netflix.graphql.dgs.internal.DgsSchemaProvider
+import com.netflix.graphql.dgs.mvc.DgsRestController
+import com.netflix.graphql.dgs.mvc.DgsRestSchemaJsonController
+import com.netflix.graphql.dgs.webmvc.autoconfigure.DgsWebMvcAutoconfiguration
+import graphql.Scalars
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLSchema
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.springframework.boot.autoconfigure.AutoConfigurations
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration
+import org.springframework.boot.autoconfigure.web.servlet.DispatcherServletAutoConfiguration
+import org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+
+class DgsWebMvcAutoConfigurationTest {
+
+    private val context = WebApplicationContextRunner().withConfiguration(
+        AutoConfigurations.of(DgsWebMvcAutoconfiguration::class.java, MockAutoConfiguration::class.java))!!
+
+    @Test
+    fun graphqlControllerAvailable() {
+        context.run { ctx ->
+            Assertions.assertThat(ctx).hasSingleBean(DgsRestController::class.java)
+        }
+    }
+
+    @Test
+    fun schemaJsonControllerAvailableWhenEnabledPropertyNotSpecified() {
+        context.run { ctx ->
+            Assertions.assertThat(ctx).hasSingleBean(DgsRestSchemaJsonController::class.java)
+        }
+    }
+
+    @Test
+    fun schemaJsonControllerAvailableWhenEnabledPropertySetToTrue() {
+        context.withPropertyValues("dgs.graphql.schema-json.enabled: true").run { ctx ->
+            Assertions.assertThat(ctx).hasSingleBean(DgsRestSchemaJsonController::class.java)
+        }
+    }
+
+    @Test
+    fun schemaJsonControllerNotAvailableWhenEnabledPropertySetToFalse() {
+        context.withPropertyValues("dgs.graphql.schema-json.enabled: false").run { ctx ->
+            Assertions.assertThat(ctx).doesNotHaveBean(DgsRestSchemaJsonController::class.java)
+        }
+    }
+
+    @Test
+    fun schemaJsonControllerMappedToDefaultPath() {
+        context.withConfiguration(AutoConfigurations.of(JacksonAutoConfiguration::class.java, WebMvcAutoConfiguration::class.java, DispatcherServletAutoConfiguration::class.java))
```

Comment:
This and the following test are new since last commit. I did not have any coverage around switching the schema-json path and it actually service at that path. This adds a bit of Fu to the context runner to allow mockmvc in the runner. Its a trick I picked up from [Spring Boot](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/integrationtest/WebMvcEndpointCorsIntegrationTests.java#L51).